src/scripts/Spells/spell_paladin.cpp

```diff
@@ -336,7 +336,7 @@ class spell_pal_ardent_defender : public SpellScriptLoader
                     // Max heal when defense skill denies critical hits from raid bosses
                     // Formula: max defense at level + 140 (raiting from gear)
                     uint32 reqDefForMaxHeal  = victim->getLevel() * 5 + 140;
-                    float pctFromDefense = (defenseSkillValue - victim->getLevel()*5.0f) / 140.0f;
+                    float pctFromDefense = (defenseSkillValue >= reqDefForMaxHeal);
```

Comment:
this is wrong:  the ``>=`` is a logical operator and returns a boolean value, which you are trying to assign to a float.